### ISSUE TITLE
Provide a warning (or error) when trying to add class attribute at runtime

### ISSUE THREAD

**Issue by hakonhagland** [rakudo/rakudo#2236]
When trying to add a class attribute at run time using the meta-object protocol, the user should get a warning that he cannot add attributes at runtime since attributes can only be modified at composition time (compile time). Here is an example:

```
use v6;

class Configuration {
}

my $config = Configuration.new;
my $attr = Attribute.new(
    :name('$.test'), # Trying to add a "test" attribute
    :type(Str),
    :has_accessor(1), 
    :package(Configuration)
);
$config.^add_attribute( $attr );  # <-- This gives no warnings/errors
$config.^compose();
say "Current attributes: ", join ', ', $config.^attributes();
$attr.set_value( $config, "Hello" ); # <-- This fails with no such attribute '$.test'
say $config.test;
```

The above program fails with the error message:
```
P6opaque: no such attribute '$.test' on type Configuration in a Configuration when trying to bind a value
  in block <unit> at ./p.p6 line 16
```
which can be confusing to a new user since the he may not realize that the error occurs since the `add_attribute()` call did not add the attribute (and did not give any warnings).

## Environment
* Operating system: Linux, Ubuntu 18.04
* Compiler version: Rakudo version 2018.04.1 built on MoarVM version 2018.04.1

**Comment by lizmat** [comment]
Feels like you need RakuAST to programmatically build classes.  So closing

**Comment by niner** [comment]
I don't think this has anything to do with RakuAST.

**Comment by lizmat** [comment]
The op wants to build a class programmatically.  RakuAST classes allow you to do that.

**Comment by jonathanstowe** [comment]
I thought the OP was suggesting that there should be a message when the programmer is attempting to add a new member attribute to an already compiled class TBH.

**Comment by lizmat** [comment]
True.  Stil the op should look at RakuAST to accomplish what they want, IMO.

Re-opened for the original request.
